### DISCUSSION TITLE
tutorials/scripting/c_sharp/c_sharp_global_classes

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#194] [User-contributed notes]
# tutorials/scripting/c_sharp/c_sharp_global_classes

Global classes (also known as named scripts) are types registered in Godot's editor so they can be used more conveniently. Global classes show up in the Add Node and Create Resource dialogs., If an...

https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_global_classes.html

**Comment by yesitsfebreeze** [comment]
Pitfall: the c# file name, must be exactly the same as the class name (case sensitive)
This will not be obviously displayed anywhere right now.

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot-docs/pull/10078

**Comment by BytesBlade** [comment]
I have added my script with the attribute mentioned but I still couldn't see my class in the Add Node window.

**Reply by snel1496** [reply]
Have you built the project? building the project is required for it to take effect in the editor. Also verify the filename matches the class name

**Comment by mpoiriert** [comment]
It would be great if we can specify a class name to use in godot

I am using namespace in C# and even if I do that i will need to make sure class name that I want to use in godot need to be unique.

Having an optional class name parameter would be great...

using Godot;

namespace Acme.A;

[GlobalClass("AcmeTestClass"]
public partial class TestClass : Node
{
}

**Reply by Calinou** [reply]
Feature proposals should be discussed on the [Godot proposals repository](https://github.com/godotengine/godot-proposals), not here.

**Comment by indiealexh** [comment]
If you are having issues using your Node or Resource, try annotating with Tool

```CSharp
[Tool, GlobalClass]
public partial class HasHealth : Resource
{
    [Export] int Health { get; set; }
    [Export] int MaxHealth { get; set; }
}
```

There is a discussion on this here: https://github.com/godotengine/godot/issues/85459

**Comment by Mloren** [comment]
Even after building project this didn't work for me until I quit and restarted Godot.
